FABulous/fabric_generator/gds_generator/flows/full_fabric_flow.py

```diff
@@ -0,0 +1,480 @@
+        (compiled_state, error_trace) for result processing
+    """
+    try:
+        from FABulous.FABulous_settings import FABulousSettings
+
+        context: FABulousSettings = init_context(project_dir=proj_dir)
+        # Reconstruct the flow in the worker process with serializable data
+        flow: FABulousTileVerilogMacroFlow = FABulousTileVerilogMacroFlow(
+            tile_type,
+            io_pin_config,
+            optimisation,
+            pdk=context.pdk,
+            pdk_root=context.pdk_root.parent,
+            base_config_path=base_config_path,
+            override_config_path=override_config_path,
+            **custom_config_overrides or {},
+        )
+        state: State = flow.start()
+    except Exception:  # noqa: BLE001
+        return None, traceback.format_exc()
+    else:
+        return state, None
+
+
+@Flow.factory.register()
+class FABulousFabricMacroFullFlow(Flow):
+    """Full automatic fabric flow with LP-optimized tile dimensions.
+
+    This flow automatically:
+    1. Compiles all tiles with 3 optimization modes to explore dimension space
+    2. Solves LP problem to find optimal dimensions minimizing fabric perimeter
+    3. Recompiles tiles with optimal dimensions from LP solution
+    4. Stitches all tiles into final fabric with minimal area
+    """
+
+    Steps = [ExtractPDKInfo, GlobalTileSizeOptimization]
+
+    config_vars = configs
+
+    def _validate_project_dir(self, proj_dir: Path, fabric: Fabric) -> None:
+        """Validate the project directory structure for required tile directories."""
+        info("Validating project directory structure...")
+        if not proj_dir.exists():
+            raise FileNotFoundError(f"Project directory not found: {proj_dir}")
+        if not proj_dir.is_dir():
+            raise NotADirectoryError(f"Project path is not a directory: {proj_dir}")
+
+        tile_dir_base: Path = proj_dir / "Tile"
+        if not tile_dir_base.exists():
+            raise FileNotFoundError(
+                f"Tile directory not found: {tile_dir_base}. "
+                "Expected structure: <proj_dir>/Tile/<tile_name>/"
+            )
+
+        # Validate all tile directories exist
+        # Check both regular tiles and SuperTiles from their dictionaries
+        missing_tiles: list[str] = []
+        found_regular_tiles: int = 0
+        found_supertiles: int = 0
+
+        # Build set of all sub-tile names that are part of SuperTiles
+        # Sub-tiles don't need their own directories
+        subtile_names: set[str] = set()
+        for supertile in fabric.superTileDic.values():
+            for subtile in supertile.tiles:
+                subtile_names.add(subtile.name)
+
+        # Validate regular tile directories
+        # Skip tiles that are sub-components of SuperTiles
+        for tile_name in fabric.tileDic:
+            if tile_name in subtile_names:
+                # This tile is part of a SuperTile, skip directory check
```

Comment:
```suggestion
                # This tile is part of a supertile, skip directory check
```

I'm a bit unsure where in this code the `SuperTile` class is meant and when just  a `supertile`, but if we just say `tile` here, we can also say `supertile`.